src/pipe_log.rs

```diff
@@ -115,60 +116,28 @@ impl Default for Version {
     }
 }
 
-#[derive(Clone, Copy, Debug, Eq, PartialEq)]
-pub enum DataLayout {
-    NoAlignment,
-    /// Alignment(block_size) mode in memory for DataLayout will make sense
-    /// when DIO is open.
-    ///
-    /// `block_size` will be dumped into the header of each log file.
-    Alignment(u64),
-}
-
-impl DataLayout {
-    pub fn from_u64(val: u64) -> Self {
-        match val {
-            0 => DataLayout::NoAlignment,
-            aligned => DataLayout::Alignment(aligned),
-        }
-    }
-
-    pub fn to_u64(self) -> u64 {
-        match self {
-            DataLayout::NoAlignment => 0,
-            DataLayout::Alignment(aligned) => {
-                debug_assert!(aligned > 0);
-                aligned
-            }
-        }
-    }
-
-    pub const fn len() -> usize {
-        std::mem::size_of::<u64>() /* serialized in u64. */
+impl Display for Version {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.to_u64().unwrap())
     }
 }
 
 #[derive(Debug, Clone)]
 pub struct LogFileContext {
     pub id: FileId,
-    pub format: LogFileFormat,
+    pub version: Version,
```

Comment:
> Hide `LogFileFormat` from outer crate, it should stay an implementation detail of file_pipe_log.

There's no need to expose it because only "file_pipe_log" needs to know about alignment.